### DISCUSSION TITLE
Possible to control fzf externally (over socket or similar)?

### DISCUSSION THREAD

**Discussion by smackesey** [junegunn/fzf#2812] [Q&A] [answered]
I'm interested in using fzf as an external "dynamic menu" that works in concert with my editor (neovim). Specifically, I'd like to run fzf _outside_ of neovim, in a separate OS window, to hold a list of diagnostics/linter errors (like an external quickfix list). The reason I want to do this is so that I can keep the list on a separate display and not disrupt my editor layout when I decide to view it.

FZF's ability to execute arbitrary commands on menu items, combined with neovim's ability to execute commands sent over a socket, solves one half of the problem-- I can control neovim from fzf. However, I'm not sure how to control fzf from neovim-- I'd like to, for example, have a key binding inside neovim for "go to next item" that both jumps to the item location in neovim _and_ changes the selection in fzf. In order to do this, I need to be able to make fzf execute arbitrary actions by sending it commands over a socket or similar. Is this possible?

**Comment by mdrv** [comment, marked as answer]
I’m aware this discussion is already old/stale, but in case someone (like me) comes here via search engine, now fzf supports `--listen <PORT>` argument to be able to receive actions externally. Related PR: https://github.com/junegunn/fzf/pull/3094
